### PR TITLE
chore: move "client" stuff to metastoreproto

### LINKED ISSUE #1128: Chore: Move metastore builtin types to sqlbuiltins crate
## Description

Move it.

I want to try to make it so that `metastore` is only imported by `glaredb`. All "client" stuff should be moved into `metastoreproto`, and all "builtin" stuff should be moved into `sqlbuiltins`.

## What This Enables

Clearer separation of concerns. Also makes it less likely to run into dependency cycles.

### PR DESCRIPTION
**Pull request by adhish20**
Fixes #1128

### CODE REVIEW

**Comment by adhish20** [comment]
sqlexec now only has a dev-dependency on metastore to run a unit test

File: crates/metastoreproto/src/errors.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        credentials: String,
+        datasource: String,
+    },
+
+    #[error("Format '{format}' not supported by datasource '{datasource}'")]
+    FormatNotSupportedByDatasource { format: String, datasource: String },
+
+    #[error("Catalog version mismatch; have: {have}, need: {need}")]
+    VersionMismtatch { have: u64, need: u64 },
+
```
Comment (vrongmeal) [inline crates/metastoreproto/src/errors.rs:23]: Not sure if it's required here.

**Review by vrongmeal — APPROVED** [review]
🚀

**Comment by vrongmeal** [comment]
@scsmithr how do you feel about renaming `metastoreproto` to `metastore_client` or something similar?

**Comment by scsmithr** [comment]
> @scsmithr how do you feel about renaming `metastoreproto` to `metastore_client` or something similar?

I'd be down with a rename.

**Comment by adhish20** [comment]
renamed metastoreproto to metastore_client

File: crates/metastoreproto/src/errors.rs (hunk 1 above)
Comment (adhish20) [reply crates/metastoreproto/src/errors.rs]: Build fails without it

**Comment by adhish20** [comment]
@scsmithr Could you merge if it looks good to you...

**Review by scsmithr — APPROVED** [review]

**Pull request merged**